vendor/github.com/OpenBazaar/wallet-interface/datastore.go

```diff
@@ -248,10 +248,33 @@ type Txn struct {
 	// This transaction only involves a watch only address
 	WatchOnly bool
 
+	// The number of confirmations on a transaction. This does not need to be saved in
+	// the database but should be calculated when the Transactions() method is called.
+	Confirmations int64
+
+	// The state of the transaction (confirmed, unconfirmed, dead, etc). Implementations
+	// have some flexibility in describing their transactions. Like confirmations, this
+	// is best calculated when the Transactions() method is called.
+	Status StatusCode
+
+	// If the Status is Error the ErrorMessage should describe the problem
+	ErrorMessage string
+
 	// Raw transaction bytes
 	Bytes []byte
 }
 
+type StatusCode string
+
+const (
+	Unconfirmed StatusCode = "UNCONFIRMED"
+	Pending                = "PENDING"
+	Confirmed              = "CONFIRMED"
+	Stuck                  = "STUCK"
+	Dead                   = "DEAD"
+	Error                  = "ERROR"
+)
```

Comment:
I like indicating what these random constants are in their name so we don't mix up wallet-interface's Error with any other Error we might want to express in this package.
`StatusUnconfirmed`, `StatusPending`, `StatusConfirmed` would be appropriate. I would also accept putting them into a status package to consume like `status.Pending`.